sdks/python/apache_beam/tools/coders_microbenchmark.py

```diff
@@ -0,0 +1,141 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+"""A microbenchmark for measuring performance of coders.
+
+This runs a sequence of encode-decode operations on random inputs
+to collect a per-element performance of various coders.
+
+Run as
+  python -m apache_beam.tools.coders_microbenchmark
+"""
+
+from __future__ import absolute_import
+from __future__ import print_function
+
+import collections
+import random
+import string
+import sys
+
+from apache_beam.coders import coders
+from apache_beam.tools import utils
+from apache_beam.transforms import window
+from apache_beam.utils import windowed_value
+
+
+def coder_benchmark_factory(name, coder, generate_fn):
+  def init(self, size):
+    self.coder_ = coder
+    self.data_ = generate_fn(size)
+
+  def call(self):
+    _ = self.coder_.decode(self.coder_.encode(self.data_))
+
+  benchmark = type(name, (object,), {})
+  benchmark.__init__ = init
+  benchmark.__call__ = call
+  return benchmark
+
+
+def generate_list_of_ints(input_size, lower_bound=0, upper_bound=sys.maxsize):
```

Comment:
"most" ints are small, but `random.randint(0, sys.maxsize)` tests the degenerate case of using all 9 bytes 99.2% of the time. Probably should use a different distribution, e.g. int(math.exp(random.random()**4 * math.log(upper_bound)). Or a different default upper_bound.

Reply:
Thanks, changed.